
Kernel sched-fair tweaks by ClearMod.


diff -uarp a/init/Kconfig b/init/Kconfig
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -140,6 +140,10 @@ config BROKEN_ON_SMP
 	depends on BROKEN || !SMP
 	default y
 
+config CLEARMOD
+	bool "Various kernel tweaks by ClearMod"
+	default y
+
 config INIT_ENV_ARG_LIMIT
 	int
 	default 32 if !UML
diff -uarp a/kernel/sched/fair.c b/kernel/sched/fair.c
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -90,7 +90,11 @@ unsigned int sysctl_sched_base_slice			=
 static unsigned int normalized_sysctl_sched_base_slice	= 750000ULL;
 #endif // CONFIG_SCHED_BORE
 
+#ifdef CONFIG_CLEARMOD
+const_debug unsigned int sysctl_sched_migration_cost	= 600000UL;
+#else
 const_debug unsigned int sysctl_sched_migration_cost	= 500000UL;
+#endif
 
 static int __init setup_sched_thermal_decay_shift(char *str)
 {
diff -uarp a/kernel/sched/sched.h b/kernel/sched/sched.h
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2591,11 +2591,19 @@ extern void deactivate_task(struct rq *r
 
 extern void wakeup_preempt(struct rq *rq, struct task_struct *p, int flags);
 
+#ifdef CONFIG_CLEARMOD
+#ifdef CONFIG_PREEMPT_RT
+# define SCHED_NR_MIGRATE_BREAK 4
+#else
+# define SCHED_NR_MIGRATE_BREAK 8
+#endif
+#else
 #ifdef CONFIG_PREEMPT_RT
 # define SCHED_NR_MIGRATE_BREAK 8
 #else
 # define SCHED_NR_MIGRATE_BREAK 32
 #endif
+#endif
 
 extern const_debug unsigned int sysctl_sched_nr_migrate;
 extern const_debug unsigned int sysctl_sched_migration_cost;
