From b0b3da03075cb7e957402b57b6bd963ebbc2b90e Mon Sep 17 00:00:00 2001
From: Peter Zijlstra <peterz@infradead.org>
Date: Tue, 26 Sep 2023 14:32:32 +0200
Subject: sched/eevdf: Allow shorter slices to wakeup-preempt

  For BORE kernel.

diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index b0cdff096736cf..64ce99cf04ec4e 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -14,6 +14,10 @@ SCHED_FEAT(PLACE_DEADLINE_INITIAL, true)
 #else // !CONFIG_SCHED_BORE
 SCHED_FEAT(RUN_TO_PARITY, true)
 #endif // CONFIG_SCHED_BORE
+/*
+ * Allow tasks with a shorter slice to disregard RUN_TO_PARITY
+ */
+SCHED_FEAT(PREEMPT_SHORT, true)
 
 /*
  * Prefer to schedule the task we woke last (assuming it failed
-- 
cgit 1.2.3-korg

